src/main/java/org/folio/services/instance/InstanceService.java

```diff
@@ -110,27 +113,53 @@ public Future<Response> updateInstance(String id, Instance newInstance) {
       });
   }
 
-  public Future<Void> deleteAllInstances() {
+  public Future<Response> deleteAllInstances() {
     return marcRepository.deleteAll()
       .compose(notUsed -> relationshipRepository.deleteAll())
       .compose(notUsed -> instanceRepository.deleteAll())
       .onSuccess(notUsed -> domainEventPublisher.publishAllRemoved())
-      .mapEmpty();
+      .map(Response.noContent().build());
   }
 
+  /**
+   * Delete instance and connected marc record.
+   */
+  // suppress "Remove useless curly braces around lambda containing only one statement"
+  @SuppressWarnings("java:S1602")
   public Future<Response> deleteInstance(String id) {
-    return instanceRepository.getById(id)
-      .compose(CommonValidators::refuseIfNotFound)
-      // before deleting the instance record delete its source marc record (foreign key!)
-      .compose(instance -> marcRepository.deleteById(id).map(notUsed -> instance))
-      .compose(instance -> {
-        final Promise<Response> deleteResult = promise();
-
-        deleteById(INSTANCE_TABLE, id, okapiHeaders, vertxContext,
-          InstanceStorage.DeleteInstanceStorageInstancesByInstanceIdResponse.class, deleteResult);
+    return instanceRepository.delete("id==" + id)
```

Comment:
Should this also check that id is safe?
That is, is it possible to inject CQL here via the `id` variable?
`String id = "c1726a23-2e7c-47c6-a278-0b177d8db9b5) or (id==*)";`

Reply:
Good catch.
CQL encoding the id string will prevent this.